src/actions/profile-view.js

```diff
@@ -45,6 +45,13 @@ export function changeSelectedThread(selectedThread: ThreadIndex): Action {
   };
 }
 
+export function changeRightClickedThread(selectedThread: ThreadIndex): Action {
```

Comment:
Unfortunately there is no simple way to feed this into the context menu, so I had to put it in Redux.

Reply:
There is the `collect` function prop to ContextMenuTrigger that can help. Using the identity function for `collect` (so this can be a static) make it possible to pass more props to `ContextMenuTrigger`, that would then be passed on to the `ContextMenu`. Or even the static function could directly pass the needed values.  Then they will be in our ContextMenu's `trigger` prop.

This is not very well documented but was introduced in https://github.com/vkbansal/react-contextmenu/pull/83 -- there's even an example there. I asked on the bug if we could have the props automatically passed if there's no `collect` defined.

So in https://github.com/devtools-html/perf.html/blob/master/src/components/header/ProfileThreadHeaderBar.js#L154 you could add a "collect" prop that would be a reference to a simple local method:
```javascript
  contextMenuCollect() {
    const { threadName, threadIndex } = props;
    return { threadName, threadIndex };
  }
```

Then you could pas this data in the `MenuItem`'s `data` prop (this one is well-documented).